ruby_programming/testing_with_rspec/introduction_to_rspec.md

```diff
@@ -0,0 +1,236 @@
+
+At the most basic level, RSpec is a Domain Specific Language written in Ruby, or, for the rest of us, a language specialized for a particular task. In this case, the task is testing Ruby code. The `rspec` gem comes packaged with all you need to get started, including five gems: `rspec-core`, `rspec-expectations`, `rspec-mocks`, `rspec-rails`, and `rspec-support`.
+
+At this point, you may be wondering, *Why RSpec? Surely, there are other frameworks out there*, and you'd be right. There are. In fact, at one point, Ruby came bundled with [Test::Unit]((https://github.com/test-unit/test-unit)) and later [Minitest](https://github.com/seattlerb/minitest) as part of its standard library, the latter of which lives on in Rails. If you tend to be pessimistic (I'm sorry, I meant *realistic*), then the [Wrong](https://github.com/sconover/wrong) testing framework might be your cup of tea. Or perhaps you're hungry and in the mood for something more substantial, in which case a side of [Bacon](https://github.com/chneukirchen/bacon) might be what you need. At the end of the day, it doesn't matter which framework you choose as long as you cultivate your testing skills. RSpec's wider adoption, especially in the Rails community, is certainly reason enough to warrant familiarity with it, but implementing tests should be the end-all and be-all, rather than choosing a particular framework.
+
+But enough proselytizing. Strap your helmet and buckle up; we're going to jump right in.
+
+### Installing RSpec
+
+Boot up your terminal and punch in `gem install rspec` to install RSpec. Once that's done, you can verify your version of RSpec with `rspec --version`, which will output the current version of each of the packaged gems. Take a minute also to hit `rspec --help` and look through the various options available.
+
+Finally, `cd` into a project directory that you wish to configure for use with RSpec and type `rspec --init` to initialize RSpec within the project. This will generate two files, `.rspec` and `spec/spec_helper.rb`, such that your project might look like:
+
+```
+project
+  |__lib
+  |   |__script.rb
+  |
+  |__spec
+  |   |__spec_helper.rb
+  |
+  |__.spec
+```
+That's it. Within two steps, you're up and running with RSpec. That was so hard, wasn't it?
+
+### Basic syntax
+How 'bout a test to see the syntax? Let's create a brand new "project" to get going. Create a new directory called "ruby_testing", change into it, and initiate RSpec.
+```
+$ mkdir ruby_testing && cd ruby_testing
+$ rspec --init
+```
+
+As expected, the output will read:
+
+```bash
+  create   .rspec
+  create   spec/spec_helper.rb
+```
+
+Run the tests from your terminal by using the `rspec` command, which will return "No examples found." That really shouldn't surprise you, because we haven't written any tests yet. If you're still shocked...maybe take a short break, or come say hello to us in our [Ruby Gitter](https://gitter.im/TheOdinProject/Ruby) room.
```

Comment:
The ruby room no longer exists - Update link to Back-End room:
`https://gitter.im/TheOdinProject/Back-End`

Reply:
:+1: